File: src/main/java/gay/skitbet/jackiro/managers/MongoManager.java
package gay.skitbet.jackiro.managers;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import gay.skitbet.jackiro.database.ServerConfigRepository;
import gay.skitbet.mongoy.Mongoy;
import lombok.Getter;

public class MongoManager {

    @Getter
    public static ServerConfigRepository serverConfigRepository;


    public static void connect(String dbName) {
        Mongoy.init(dbName);
        initRepositories();
    }

    public static void initRepositories() {
        MongoManager.serverConfigRepository = new ServerConfigRepository(); // setup repo
    }

}

File: src/main/kotlin/gay/skitbet/jackiro/managers/MongoManager.kt
package gay.skitbet.jackiro.managers

import gay.skitbet.jackiro.database.ServerConfigRepository
import gay.skitbet.mongoy.Mongoy

object MongoManager {

    val serverConfigRepository: ServerConfigRepository by lazy { ServerConfigRepository() }

    fun connect(dbName: String) {
        Mongoy.init(dbName)
        initRepositories()
    }

    private fun initRepositories() {
        // Repositories are initialized lazily via the serverConfigRepository property
    }
}
